src/Microsoft.DotNet.Darc/src/DarcLib/RemoteRepoBase.cs

```diff
@@ -0,0 +1,158 @@
+                        {
+                            AllowEmptyCommit = false,
+                            PrettifyMessage = true
+                        });
+
+                        localRepo.Network.Push(localRepo.Branches[branch], new LibGit2Sharp.PushOptions
+                        {
+                            CredentialsProvider = (url, user, cred) =>
+                            new LibGit2Sharp.UsernamePasswordCredentials
+                            {
+                                // The PAT is actually the only thing that matters here, the username
+                                // will be ignored.
+                                Username = dotnetMaestro,
+                                Password = pat
+                            }
+                        });
+                    }
+                }
+                catch (LibGit2Sharp.EmptyCommitException)
+                {
+                    _logger.LogInformation("There was nothing to commit...");
+                }
+                catch (Exception exc)
+                {
+                    throw new DarcException($"Something went wrong when pushing the files to repo {repoUri} in branch {branch}", exc);
+                }
+                finally
+                {
+
+                    try
+                    {
+                        // Libgit2Sharp behaves similarly to git and marks files under the .git/objects hierarchy as read-only, 
+                        // thus if the read-only attribute is not unset an UnauthorizedAccessException is thrown.
+                        GitFileManager.NormalizeAttributes(tempRepoFolder);
+
+                        Directory.Delete(tempRepoFolder, true);
+                    }
+                    catch (DirectoryNotFoundException)
+                    {
+                        // If the directory wasn't found, that means that the clone operation above failed
+                        // but this error isn't interesting at all.
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Decode a base64 string or return the original string if not in base 64 form.
+        /// </summary>
+        /// <param name="encodedContent">Potentially base64 encoded string</param>
+        /// <returns>Decoded string as utf8</returns>
+        protected string GetDecodedContent(string encodedContent)
+        {
+            try
+            {
+                byte[] content = Convert.FromBase64String(encodedContent);
+                return Encoding.UTF8.GetString(content);
+            }
+            catch (FormatException)
+            {
+                return encodedContent;
+            }
+        }
+
+        protected byte[] GetContentBytes(string content)
+        {
+            string decodedContent = GetDecodedContent(content);
+            return Encoding.UTF8.GetBytes(decodedContent);
```

Comment:
>Encoding.UTF8.GetBytes(decodedContent); [](start = 19, length = 39)

... Does this mean we encode it into UTF-8 in the above method... and then decode it back? Why not just return the Convert.FromBase64String?  What if the base 64 string isn't valid UTF-8 for some reason? Would we expect this method to fail to round trip the encoded byte array?

Reply:
Not exactly sure.  I just copied this over when moving code into the base class.  I'll take a closer look.